TypeCobol.LanguageServer/WorkspaceProjectStore.cs

```diff
@@ -0,0 +1,347 @@
+        /// <param name="defaultProjectKey">The key of the default workspace project</param>
+        /// <param name="useAntlrProgramParsing">Using ANTLR parsing or not</param>
+        /// <param name="useEuroInformationLegacyReplacingSyntax">Use E-I options or not</param>
+        internal WorkspaceProjectStore(string rootDirectoryFullName, string defaultProjectKey, bool useAntlrProgramParsing, bool useEuroInformationLegacyReplacingSyntax)
+        {
+            this._rootDirectoryFullName = rootDirectoryFullName;
+            this._allOpenedDocuments = new ConcurrentDictionary<Uri, DocumentContext>();
+            this._workspaceProjects = new ConcurrentDictionary<string, WorkspaceProject>(StringComparer.OrdinalIgnoreCase);
+            CreateDefaultWorkspaceProject(defaultProjectKey, useAntlrProgramParsing, useEuroInformationLegacyReplacingSyntax);
+        }
+
+        /// <summary>
+        /// The default WorkspaceProject instance.
+        /// </summary>
+        internal WorkspaceProject DefaultWorkspaceProject { get; private set; }
+
+        /// <summary>
+        /// Create the default workspace project
+        /// </summary>
+        /// <param name="projectKey"></param>
+        /// <param name="useAntlrProgramParsing"></param>
+        /// <param name="useEuroInformationLegacyReplacingSyntax"></param>
+        private void CreateDefaultWorkspaceProject(string projectKey, bool useAntlrProgramParsing, bool useEuroInformationLegacyReplacingSyntax)
+        {
+            var defaultDocumentFormat = new DocumentFormat(Encoding.GetEncoding("iso-8859-1"), EndOfLineDelimiter.CrLfCharacters, 80, ColumnsLayout.CobolReferenceFormat);
+            CompilationProject compilationProject = new CompilationProject(
+                projectKey, _rootDirectoryFullName, Helpers.DEFAULT_EXTENSIONS, defaultDocumentFormat,
+                new TypeCobolOptions(), null); //Initialize a default CompilationProject - has to be recreated after ConfigurationChange Notification
+            compilationProject.CompilationOptions.UseAntlrProgramParsing =
+                compilationProject.CompilationOptions.UseAntlrProgramParsing || useAntlrProgramParsing;
+
+            compilationProject.CompilationOptions.UseEuroInformationLegacyReplacingSyntax =
+                compilationProject.CompilationOptions.UseEuroInformationLegacyReplacingSyntax ||
+                useEuroInformationLegacyReplacingSyntax;
+
+            // Create the Default Workspace Project instance.
+            DefaultWorkspaceProject = new WorkspaceProject(projectKey, compilationProject);
+            // Add it in the store
+            this.AddProject(DefaultWorkspaceProject);
+        }
+
+        /// <summary>
+        /// Indicates whether this workspace project store has opened documents or not.
+        /// </summary>
+        internal bool IsEmpty => this._workspaceProjects.Values.All(wksPrj => wksPrj.IsEmpty);
```

Comment:
No, use the dictionary of opened documents. And you can remove the `IsEmpy` property on `WorkspaceProject`